### PR TITLE
feat(RHINENG-11548): migrate report detail header to REST

### PR DESCRIPTION
**Pull request by mkholjuraev**
## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist
Migrates report detail header to REST api with flags. Due to ID discrepency, you need to find some valid ID from the new REST api and then access the page to test.

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: src/SmartComponents/ReportDetails/ReportDetails.js [hunk 1]
```diff
@@ -195,8 +164,66 @@ export const ReportDetails = ({ route }) => {
...
+
+  return (
+    <ReportDetailsBase
+      {...{ route, data: data?.profile, error, loading, ssgVersions }}
+    />
+  );
+};
+
+ReportDetailsGrapgQL.propTypes = {
   route: propTypes.object,
```
Comment (gkarat) [inline src/SmartComponents/ReportDetails/ReportDetails.js:201]:
```suggestion
  route: propTypes.object.shape({
    title: PropTypes.string.isRequired,
    defaultTitle: PropTypes.string.isRequired
  }).isRequired,
```

File: src/SmartComponents/ReportDetails/ReportDetails.js [hunk 2]
```diff
@@ -195,8 +164,66 @@ export const ReportDetails = ({ route }) => {
...
+
+  return (
+    <ReportDetailsBase
+      {...{ route, data: dataSerialiser(data, dataMap), error, loading }}
+    />
+  );
+};
+
+ReportDetailsRest.propTypes = {
+  route: propTypes.object,
```
Comment (gkarat) [inline src/SmartComponents/ReportDetails/ReportDetails.js:216]:
```suggestion
  route: propTypes.object.shape({
    title: PropTypes.string.isRequired,
    defaultTitle: PropTypes.string.isRequired
  }).isRequired,
```

File: src/SmartComponents/ReportDetails/ReportDetails.js [hunk 3]
```diff
@@ -195,8 +164,66 @@ export const ReportDetails = ({ route }) => {
...
-ReportDetails.propTypes = {
+ReportDetailsBase.propTypes = {
+  route: propTypes.object,
+  data: propTypes.object,
+  error: propTypes.object,
+  loading: propTypes.bool,
+  ssgVersions: propTypes.array,
+};
+
+//depricated component
```
Comment (gkarat) [inline src/SmartComponents/ReportDetails/ReportDetails.js:175]:
```suggestion
// deprecated component
```

File: src/SmartComponents/ReportDetails/ReportDetails.js [hunk 4]
```diff
@@ -195,8 +164,66 @@ export const ReportDetails = ({ route }) => {
...
+const ReportDetailsGrapgQL = ({ route }) => {
```
Comment (gkarat) [inline src/SmartComponents/ReportDetails/ReportDetails.js:176]:
```suggestion
const ReportDetailsGraphQL = ({ route }) => {
```

File: src/SmartComponents/ReportDetails/ReportDetails.js [hunk 5]
```diff
@@ -31,62 +31,31 @@ import './ReportDetails.scss';
...
-    ssgVersions = [
-      ...new Set(
-        profile.policy.profiles.flatMap(({ benchmark: { version } }) => version)
-      ),
-    ];
   }
 
   useTitleEntity(route, policyName);
 
+  console.log('hello data', data);
```
Comment (gkarat) [inline src/SmartComponents/ReportDetails/ReportDetails.js:58]:
```suggestion
```

File: src/SmartComponents/ReportDetails/ReportDetails.test.js [hunk 6]
```diff
@@ -1,8 +1,13 @@
 import { render, screen } from '@testing-library/react';
 import '@testing-library/jest-dom';
 import TestWrapper from '@/Utilities/TestWrapper';
-import { QUERY, ReportDetails } from './ReportDetails';
+import ReportsDetail from './ReportDetails';
```
Comment (gkarat) [inline src/SmartComponents/ReportDetails/ReportDetails.test.js:4]:
```suggestion
import ReportDetails from './ReportDetails';
```

File: src/SmartComponents/ReportDetails/ReportDetails.test.js [hunk 7]
```diff
@@ -77,3 +86,26 @@ describe('ReportDetails', () => {
     ).toBeInTheDocument();
```
Comment (gkarat) [inline src/SmartComponents/ReportDetails/ReportDetails.test.js:86]: Perhaps we could one extra test ensuring that the GQL function is called and that the component is doing the right request according to the feature flag?

File: src/SmartComponents/ReportDetails/ReportDetails.test.js [hunk 8]
```diff
@@ -77,3 +86,26 @@ describe('ReportDetails', () => {
...
 });
+
+describe('Report Details - REST', () => {
+  beforeEach(() => {
+    useAPIV2FeatureFlag.mockImplementation(() => true);
+  });
+
+  it('should use REST api', () => {
+    useReport.mockImplementation(() => ({
+      data: [],
```
Comment (gkarat) [inline src/SmartComponents/ReportDetails/ReportDetails.test.js:97]: Can we have another test with non-empty data and assert against the rendered content?

**Review by gkarat — COMMENTED** [review]
Rather minor notes, overall looks good

File: src/SmartComponents/ReportDetails/ReportDetails.js (hunk 5 above)
Comment (bastilian) [reply src/SmartComponents/ReportDetails/ReportDetails.js:58]: 👋

**Comment by mkholjuraev** [comment]
@gkarat @bastilian now all suggestions are addressed

**Review by AsToNlele — COMMENTED** [review]
Code looks good!
Yeah the detail needs to be fixed, provided you a quick solution.
The Business objective is broken, because the serializer returns `businessObjective: {title: null}` this can be fixed in `ReportDetailsDescription.js` on line 47 by adding `?.title` and will work on both of the versions of the API

**Comment by mkholjuraev** [comment]
@AsToNlele thank you for the fix! Now, it is ready to review again

**Review by gkarat — APPROVED** [review]
Awesome, great job @mkholjuraev
